### PR TITLE
Challenge PR

### PR DESCRIPTION
**Pull request by lauri-neuvonen**
First task pull request.

### CODE REVIEW

File: gizmo.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# First draft for gizmo.py
+import numpy as np
+import pandas as pd
+
+def hello(name, country='Finland'):
+    print('Hello ' + name + ', how are things in ' + country + '?')
```
Comment (wmvanvliet) [inline gizmo.py:6]: [f-strings](https://realpython.com/python-f-strings/) are awesome for this type of work:

```python
print(f'(Hello {name}, how are things in {country}?')
```

File: gizmo.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+def spell(word):
+
+    output = ""
+    for i,l in enumerate(word):
+        if i == len(word)-1:
+            output = output + l
```
Comment (wmvanvliet) [inline gizmo.py:13]: alternatively:

```python
output += l
```

File: gizmo.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    print(output)
+
+def relative_path(ids):
+
+    list = []
+    path_base = './subjects/mock_recording_'
+    suffix = '.rec'
+    for id in ids:
+        list.append(path_base + id + suffix)
```
Comment (wmvanvliet) [inline gizmo.py:25]: Here, [f-strings](https://realpython.com/python-f-strings/) are a clear win:

```python
list.append(f'./subjects/mock_recordings_{id}.rec')
```

much more readable than needing separate variables for `path_base` and `suffix`.

File: gizmo.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        m_table = np.outer(nums, nums)
+        m_table[m_table % zero_out_multiples == 0] = 0
+        return m_table
+
+def generate_fibonacci_sequence(n):
+
+
+    if n == 1:
+        seq = [0]
+        yield seq[-1]
```
Comment (wmvanvliet) [inline gizmo.py:62]:
```suggestion
        yield 0
```

File: gizmo.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        seq = [0, 1]
+        yield 0
+        yield 1
+        for i in range(1, n-1):
+            seq.append(seq[i-1] + seq[i])
+            yield seq[-1]
+
+def get_fibonacci_sequence(n):
+    res_array = np.empty(n)
+    for i, el in enumerate(generate_fibonacci_sequence(n)): res_array[i] = int(el)
```
Comment (wmvanvliet) [inline gizmo.py:78]: a loop like this is really slow. Here is a faster alternative:

```python
return np.array(list(generate_fibonacci_sequence(n)))
```

This has the downside of copying the data once, but the upside of not using a Python loop.

**Review by wmvanvliet — APPROVED** [review]
Well done! I've added some comments and nitpicks. Take a look at [f-strings](https://realpython.com/python-f-strings/), they will blow your mind! Nice work with the multiplication table.

File: gizmo.py (hunk 5 above)
Comment (lauri-neuvonen) [reply gizmo.py:78]: Thanks for all the comments! F-strings are definitely useful. 
-Lauri

**Pull request closed without merging**